src/openrct2-ui/windows/NewRide.cpp

```diff
@@ -1044,19 +1039,18 @@ static void window_new_ride_select(rct_window *w)
     ride_construct_new(item);
 }
 
-static void window_new_ride_list_vehicles_for(const uint8 rideType, const rct_ride_entry * rideEntry, char * out)
+static void window_new_ride_list_vehicles(const uint8 rideType, const rct_ride_entry * rideEntry, rct_drawpixelinfo * dpi, sint32 x, sint32 y)
 {
     rct_ride_entry * currentRideEntry;
     const RideGroup * rideGroup, * currentRideGroup;
     sint32 rideEntryIndex;
 
     if (RideGroupManager::RideTypeIsIndependent(rideType))
     {
-        out[0] = 0;
         return;
     }
 
-    memset(out, 0, AVAILABILITY_STRING_SIZE);
+    char * availabilityString = (char *)calloc(AVAILABILITY_STRING_SIZE, sizeof(char));
```

Comment:
char availabilityString [AVAILABILITY_STRING_SIZE] = {};

Reply:
That causes crashes. It was the reason for using `strdup` previously. Only a pointer works.

Reply:
Where would it actually crash?

Reply:
In the gfx_draw_string_left_clipped function, because it uses illegal memory.